From 2e6870993d226dd8af3e2db502e8e183ee63d66a Mon Sep 17 00:00:00 2001
From: Alexey Kardashevskiy <aik@ozlabs.ru>
Date: Wed, 21 May 2014 18:16:01 +1000
Subject: [PATCH] trace: Replace error with warning if event is not defined

At the moment QEMU exits if trace point is not defined which makes
a developer life harder if he has to switch between branches with
different traces implemented.

This replaces error+exit wit WARNING if the tracepoint does not exist or
not traceable.

Signed-off-by: Alexey Kardashevskiy <aik@ozlabs.ru>
Signed-off-by: Stefan Hajnoczi <stefanha@redhat.com>
(cherry picked from commit 82432638ebeedda8a2e18838b6fbef4b14a94f31)
---
 trace/control.c | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/trace/control.c b/trace/control.c
index 49f61e1..4aa02cf 100644
--- a/trace/control.c
+++ b/trace/control.c
@@ -112,15 +112,15 @@ void trace_backend_init_events(const char *fname)
                 TraceEvent *ev = trace_event_name(line_ptr);
                 if (ev == NULL) {
                     fprintf(stderr,
-                            "error: trace event '%s' does not exist\n", line_ptr);
-                    exit(1);
-                }
-                if (!trace_event_get_state_static(ev)) {
+                            "WARNING: trace event '%s' does not exist\n",
+                            line_ptr);
+                } else if (!trace_event_get_state_static(ev)) {
                     fprintf(stderr,
-                            "error: trace event '%s' is not traceable\n", line_ptr);
-                    exit(1);
+                            "WARNING: trace event '%s' is not traceable\n",
+                            line_ptr);
+                } else {
+                    trace_event_set_state_dynamic(ev, enable);
                 }
-                trace_event_set_state_dynamic(ev, enable);
             }
         }
     }
